### ISSUE TITLE
Capture type of Emphasis/Strong in AST

### ISSUE THREAD

**Issue by jfrolich** [ocaml/omd#214]
`Star` or `Underscore`. This allows parsers to treat them slightly different.

**Comment by nojb** [comment]
Actually we supported this before and I removed it because I thought it was not useful. But I'm planning to put it back since it turns out it is useful for a number of things.

**Comment by nojb** [comment]
@jfrolich On further thought could you say a few words on what use-case you have in mind for this?

**Comment by jfrolich** [comment]
For compilation to ocamldoc which has 3 emphasis/strong/bold styles.

**Comment by nojb** [comment]
You mean `{e text}`, `{b text}` and `{i text}` ?

**Comment by jfrolich** [comment]
Yes. I figured we could stay in the markdowns spec and still support these differences.

**Comment by nojb** [comment]
I see. I am a bit reticent to adding an extra field to the AST that will be ignored most of time... On the other hand, this may come in handy for the hypothetical markdown backend, so it may be worth it after all... I will give it a shot, but am not fully convinced yet :)
